### ISSUE TITLE
Generic API entry point

### ISSUE THREAD

**Issue by aguibert** [eclipse-vertx/vertx-sql-client#616]
#### Describe the feature

In order for other projects (such as Hibernate Reactive) to build on top of the generic vertx-sql-client API, they need a mechanism to initialize a connection/client/pool without a direct dependency on the DB-specific API.

For example, Hibernate Reactive has to do stuff like this:
```java
switch ( uri.getScheme() ) {
	case "postgresql":
		PgConnectOptions pgOptions = new PgConnectOptions()
				.setPort( uri.getPort() )
				.setHost( uri.getHost() )
				.setDatabase( database )
				.setUser( username );
		if (password != null) {
			pgOptions.setPassword( password );
		}
		this.pool = PgPool.pool(Vertx.vertx(), pgOptions, poolOptions);
		break;
	case "mysql":
		MySQLConnectOptions mysqlOptions = new MySQLConnectOptions()
				.setPort( uri.getPort() )
				.setHost( uri.getHost() )
				.setDatabase( database )
				.setUser( username );
		if (password != null) {
			mysqlOptions.setPassword( password );
		}
		this.pool = MySQLPool.pool(Vertx.vertx(), mysqlOptions, poolOptions);
		break;
}
```

With JDBC, each driver is discoverable via the JDK ServiceLoader mechanism, so that no driver-specific API is needed in order to integrate with the driver.

Proposed changes:
- Define a new `io.vertx.sqlclient.Driver` interface, used for creating `Pool` objects given `SqlConnectOptions` objects
- Make `io.vertx.sqlclient.SqlConnectOptions` a concrete class so it can be directly instantiated
- Add static `pool()` methods to `io.vertx.sqlclient.Pool` that detect the Pool implementation from the ServiceLoader. An error is thrown if 0 or > 1 implementations are found
- Add `META-INF/service/io.vertx.sqlclient.Driver` files to each driver

#### Use cases

- Up-stack technologies can more easily integrate with vertx-sql-client if they can just depend on the generic `vertx-sql-client` artifact instead of needing to depend on each specific driver in order to initialize connections/pools.

#### Contribution

I can implement this feature

**Comment by gavinking** [comment]
Agreed, I was thinking this too.

**Comment by aguibert** [comment]
Closed by #618
